common/string_helpers.cpp

```diff
@@ -81,8 +82,40 @@ auto UnescapeStringLiteral(llvm::StringRef source, const int hashtag_num,
           ret.push_back(16 * *c1 + *c2);
           break;
         }
-        case 'u':
-          CARBON_FATAL() << "\\u is not yet supported in string literals";
+        case 'u': {
+          i++;
+          if (i >= source.size() || source[i] != '{')
+            return std::nullopt;
```

Comment:
```suggestion
          if (i >= source.size() || source[i] != '{') {
            return std::nullopt;
          }
```

Just a matter of style -- [we always use braces for conditionals](https://github.com/carbon-language/carbon-lang/blob/trunk/docs/project/cpp_style_guide.md#syntax-and-formatting).